app/src/org/commcare/android/storage/framework/Persisted.java

```diff
@@ -98,6 +97,7 @@ public int getID() {
     }
 
     private void readVal(Field f, Object o, DataInputStream in) throws DeserializationException, IOException, IllegalAccessException {
+        synchronized (fieldOrderings) {
```

Comment:
Ah, realized that if we assume that `Class.getDeclaredFields()` always returns the same `Field` instances, then we can synchronize over them. Did this in 
https://github.com/dimagi/commcare-odk/pull/1303/commits/4204c98579f9f1b5c3406a370fb98dcba697881b , should be good to go.